internal/blobstore/blobstore.go

```diff
@@ -292,8 +292,9 @@ func NewRefs(n int) *Refs {
 // Add records that the given hash is referenced.
 // It ignores the hash if it's invalid.
 func (r *Refs) Add(hash string) {
-	data, ok := decodeHash(hash)
-	if !ok {
+	data, err := decodeHash(hash)
+	if err != nil {
+		logger.Errorf("cannot add bad hash %q: %v", hash, err)
```

Comment:
@mhilton We added a check to not use add when it's empty in here https://github.com/juju/charmstore/pull/764/files#diff-20be4fa0bfcb7416de93e699d65b2d2aR459